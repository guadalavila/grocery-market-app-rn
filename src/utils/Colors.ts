export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
