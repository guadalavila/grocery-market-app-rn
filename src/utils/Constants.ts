export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
