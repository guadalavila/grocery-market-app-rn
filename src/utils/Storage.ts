import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
