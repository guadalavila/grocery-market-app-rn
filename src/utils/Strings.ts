export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
