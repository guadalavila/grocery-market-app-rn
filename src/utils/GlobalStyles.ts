import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
