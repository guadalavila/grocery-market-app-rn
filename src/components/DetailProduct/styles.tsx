import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    divider: {
        width: 70,
        height: 8,
        backgroundColor: Colors.divider,
        borderRadius: 20,
        marginVertical: Sizes.M,
        alignSelf: 'center',
    },
});
