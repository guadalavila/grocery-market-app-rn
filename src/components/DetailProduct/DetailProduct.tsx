import React from 'react';
import { Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { flex, GlobalStyles } from '../../utils/GlobalStyles';
import { styles } from './styles';
import { Colors } from '../../utils/Colors';
import { TouchableOpacity } from 'react-native-gesture-handler';
import ButtonPrimary from '../ButtonPrimary/ButtonPrimary';

const DetailProduct = () => {
    return (
        <View style={flex(1)}>
            <View style={flex(5)}>
                <View style={styles.divider} />
                <View style={styles.container}>
                    <View style={[GlobalStyles.row, GlobalStyles.justBtw, styles.buttons]}>
                        <View style={styles.containerPrice}>
                            <Text style={styles.priceText}>$ 220,33</Text>
                        </View>
                        <TouchableOpacity activeOpacity={0.7} onPress={() => {}} style={styles.containerFav}>
                            <Icon size={22} name='heart' color={Colors.red} />
                        </TouchableOpacity>
                    </View>
                    <Text style={styles.header}>Bananas Organicas</Text>
                    <View style={styles.containerDesc}>
                        <Text style={styles.descText}>
                            It is a long established fact that a reader will be distracted by the readable content of a
                            page when looking at its layout. The point of using Lorem Ipsum is that it has a
                            more-or-less normal distribution of letters, as opposed to using 'Content here, content
                            here', making it look like readable English.
                        </Text>
                    </View>
                </View>
            </View>
            <View style={flex(1)}>
                <ButtonPrimary label='Agregar al Carrito' onPress={() => {}} />
            </View>
        </View>
    );
};

export default DetailProduct;
