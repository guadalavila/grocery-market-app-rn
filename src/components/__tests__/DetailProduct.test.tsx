import { render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import DetailProduct from '../DetailProduct/DetailProduct';

describe('test DetailProduct component', () => {
    const productMock = {
        id: '6b17d300-943a-11ed-a1eb-0242ac120002',
        image: 'https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp',
        name: 'Banana',
        description: 'Organicas',
        price: 20.0,
    };
    const onPressMock = jest.fn();

    test('should match with snapshot', () => {
        const component = render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(component).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(screen).toBeTruthy();
    });
    test('should render all texts', () => {
        render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByText(productMock.name)).toBeTruthy();
        expect(screen.getByText(`$ ${productMock.price}.00`)).toBeTruthy();
    });

    test('should render icon', () => {
        render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByTestId('icon-heart')).toBeTruthy();
        expect(screen.getByTestId('icon-heart').props.name).toBe('heart-outline');
        expect(screen.getByTestId('icon-heart').props.size).toBe(22);
        expect(screen.getByTestId('icon-heart').props.color).toBe(Colors.red);
    });

    test('should render button heart', () => {
        render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByTestId('button-heart')).toBeTruthy();
    });

    test('should render add to cart button', () => {
        render(<DetailProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByTestId('button-add-to-cart')).toBeTruthy();
    });
});
