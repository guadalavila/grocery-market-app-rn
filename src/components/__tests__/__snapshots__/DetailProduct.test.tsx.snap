// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DetailProduct component should match with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EFEEF1",
          "borderRadius": 20,
          "height": 8,
          "marginVertical": 16,
          "width": 70,
        }
      }
    />
    <View
      style={
        Object {
          "marginHorizontal": 24,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
            Object {
              "marginBottom": 16,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#DEFAE8",
              "borderRadius": 20,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#68B984",
                "fontWeight": "bold",
              }
            }
          >
            $ 
            20
            .00
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFE8E8",
              "borderRadius": 20,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="button-heart"
        >
          <MockedIoniconsIcons
            color="#FF1C1E"
            name="heart-outline"
            size={22}
            testID="icon-heart"
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 26,
            "fontWeight": "bold",
          }
        }
      >
        Banana
      </Text>
      <View
        style={
          Object {
            "marginVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#9f9f9f",
              "fontSize": 15,
              "fontWeight": "500",
            }
          }
        >
          It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English.
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#68B984",
          "borderRadius": 12,
          "marginHorizontal": 10,
          "opacity": 1,
          "padding": 12,
        }
      }
      testID="button-add-to-cart"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Agregar al carrito
      </Text>
    </View>
  </View>
</View>
`;
